### PR TITLE
update PFCluster calibration for HGCal v9 geometry

### PR DESCRIPTION
**Pull request by clelange**
Details of the calibration procedure can be taken from the followings sets of slides presented at the [HGCAL DPG simulation and performance meeting](https://indico.cern.ch/event/776790/) on 28th November:

- [photons](https://indico.cern.ch/event/776790/)
- [pions](https://indico.cern.ch/event/776790/contributions/3230584/attachments/1761518/2858298/20181128_HGCalDPG_PFCluster_PionScaleResolution_D28.pdf)

The methodology is the same for both and has not changed w.r.t. the previous iteration (but with the old geometry). The final results tables (using particles with pT = 50 GeV) can be found on page 8 of the respective presentations.

### CODE REVIEW

**Comment by perrotta** [comment]
please test

**Comment by kpedro88** [comment]
assign upgrade

**Comment by kpedro88** [comment]
@fabiocos @smuzaffar @cvuosalo why does this PR have a "Codacy/PR Quality Review" check listed?

**Comment by smuzaffar** [comment]
@davidlange6  was looking in to codacy that might have enabled these checks

File: RecoParticleFlow/PFClusterProducer/python/particleFlowRealisticSimClusterHGCCalibrations_cfi.py [hunk 1]
```diff
@@ -2,5 +2,5 @@
 
 minEtaCorrection = cms.double(1.4)
 maxEtaCorrection = cms.double(3.0)
-hadronCorrections = cms.vdouble(1.24, 1.24, 1.24, 1.23, 1.24, 1.25, 1.29, 1.29)  
-egammaCorrections = cms.vdouble(1.00, 1.00, 1.01, 1.01, 1.02, 1.03, 1.04, 1.04)  
+hadronCorrections = cms.vdouble(1.28, 1.28, 1.24, 1.19, 1.17, 1.17, 1.17, 1.17)
```
Comment (kpedro88) [inline RecoParticleFlow/PFClusterProducer/python/particleFlowRealisticSimClusterHGCCalibrations_cfi.py]: if these calibrations are only appropriate for the V9 geometry, the values should be modified using the `phase2_hgcalV9` modifier, so the old versions of HGCal will not get incorrect calibrations

**Comment by clelange** [comment]
Hi @kpedro88 - yes, this is only v9. I've never used modifiers before, so I'm not 100% sure I use them correctly here. Should the implementation be incorrect, could you please tell me how to fix it (i.e. do I have to import anything from https://github.com/cms-sw/cmssw/blob/master/RecoParticleFlow/PFClusterProducer/python/particleFlowClusterHGC_cfi.py and modify things in there instead)?

**Comment by kpedro88** [comment]
see https://github.com/clelange/cmssw/pull/1

**Comment by kpedro88** [comment]
please test

**Comment by davidlange6** [comment]
it seems I did without realizing.. They should be back off again (presumably for future PRs) until we get some more experience with this thing being useful or not

**Comment by fabiocos** [comment]
@davidlange6 it looks useful at a glance, but I think we should define which kind of problems we want to track or enforce to be fixed

**Comment by perrotta** [comment]
+1
- New calibration applied to phase 2 hgcal V9 only, via the ```phase2_hgcalV9``` era
- Calibration results linked in the PR description for photons and pions have been shown to improve scale and resolution
- Jenkins tests pass and show modifications only in the affected workflow which contains the 2023D35 geometry

**Comment by kpedro88** [comment]
+upgrade

**Comment by fabiocos** [comment]
+1

**Pull request merged**